api/src/controllers/AuthController.ts

```diff
@@ -60,19 +64,38 @@ export class AuthController {
     }
     @HttpPost('/provider')
     public async providerLogin(
-        @BodyParam('provider') provider: keyof typeof providers,
+        @BodyParam('provider') provider: string,
         @BodyParam('authToken') authToken: string
     ) {
-        const providerUrl = providers[provider].url;
-        return request({
-            url: providerUrl,
-            qs: {access_token: authToken}
-        }, async (error, response, body) => {
-            if (!error && response.statusCode == 200) {
-                return await this.authService.registerWithProvider(body)
-            } else {
-                throw new BadRequestError(error);
-            }
+        try {
+            const response = await this.validateWithProvider(
+                provider,
+                authToken
+            )
+            return await this.authService.registerWithProvider(response)
+        } catch (e) {
+            throw new BadRequestError('Invalid token')
+        }
+    }
+
+    validateWithProvider(network, socialToken) {
+        return new Promise(function(resolve, reject) {
+            request(
+                {
+                    url: providers[network].url,
+                    qs: {
+                        access_token: socialToken,
+                        fields: 'email, first_name, last_name, birthday',
```

Comment:
Put the fields in social providers config file, for each social provider

Reply:
done